modules/serverless-kafka-broker-sasl-default-config.adoc

```diff
@@ -0,0 +1,96 @@
+// Module is included in the following assemblies:
+//
+// * serverless/knative_eventing/serverless-kafka.adoc
+
+[id="serverless-kafka-broker-sasl-default-config_{context}"]
+= Configuring SASL authentication for the Kafka Broker
+
+.Prerequisites
+
+* A username and password for the Kafka cluster.
+* Choose the SASL mechanism to use, for example `PLAIN`, `SCRAM-SHA-256`, or `SCRAM-SHA-512`.
+* If TLS is enabled, you also need the `ca.crt` certificate file for the Kafka cluster.
+
+[NOTE]
+====
+Red Hat recommends to enable TLS in addition to SASL.
+====
+
+.Procedure
+
+. Create the certificate files as secrets in the `knative-eventing` namespace:
++
+[source,terminal]
+----
+$ oc create secret --namespace knative-eventing generic <kafka_auth_secret> \
+  --from-literal=protocol=SASL_SSL \
+  --from-literal=sasl.mechanism=<sasl_mechanism> \
+  --from-file=ca.crt=caroot.pem \
+  --from-literal=password="SecretPassword" \
+  --from-literal=user="my-sasl-user"
+----
++
+[IMPORTANT]
+====
+Use the key names `ca.crt`, `password`, and `sasl.mechanism`. Do not change them.
+====
+
+. Start editing the `KnativeKafka` custom resource:
++
+[source,terminal]
+----
+$ oc edit knativekafka
+----
+
+. Reference your secret and the namespace of the secret:
++
+[source,yaml]
+----
+apiVersion: operator.serverless.openshift.io/v1alpha1
+kind: KnativeKafka
+metadata:
+  namespace: knative-eventing
+  name: knative-kafka
+spec:
+  broker:
+    enabled: true
+    defaultConfig:
+      bootstrapServers: <bootstrap_servers>
+      numPartitions: <num_partitions>
+      replicationFactor: <replication_factor>
+      authSecretName: <kafka_auth_secret>
+  channel:
+    enabled: false
+  source:
+    enabled: false
+----
++
+[NOTE]
+====
+Make sure to specify the matching port in the bootstrap server.
+====
++
+For example:
++
+[source,yaml]
+----
+apiVersion: operator.serverless.openshift.io/v1alpha1
+kind: KnativeKafka
+metadata:
+  namespace: knative-eventing
+  name: knative-kafka
+spec:
+  broker:
+    enabled: true
+    defaultConfig:
+      bootstrapServers: eventing-kafka-bootstrap.kafka.svc:9093
+      authSecretName: scram-user
+  channel:
+    enabled: false
+  source:
+    enabled: false
+----
+
+.Additional resources
+
+* link:https://access.redhat.com/documentation/en-us/red_hat_amq/7.5/html-single/using_amq_streams_on_rhel/index#assembly-kafka-encryption-and-authentication-str[TLS and SASL on Kafka]
```

Comment:
See comments on other sections re links and prereqs

Reply:
Both `.Prerequisites` and `.Additional resources` are copied from the `modules/serverless-kafka-sasl.adoc` file, as I used that as inspiration for this subject